streaming-payments.md

```diff
@@ -0,0 +1,375 @@
+    console.log('user! writing head', user)
+    res.writeHead(200, {
+      'Pay': [ 1, plugin.getAccount() + '.' + user, base64(secret) ].join(' ')
+    })
+    // Flush the headers in a first TCP packet:
+    res.socket.write(res._header)
+    res._headerSent = true
+  }).listen(8000)
+})
+```
+
+To run it, use:
+
+```sh
+npm install interledgerjs/ilp-plugin-payment-channel-framework#mj-fix
+node ./shop3.js
+```
+
+And here is the content consumption tool, which first transfers 1000 XRP to its account at the shop's ledger, and then
+consumes 1000 letters. Save it as `tool2.js`:
+```js
+const IlpPacket = require('ilp-packet')
+const Plugin = require('ilp-plugin-payment-channel-framework')
+const crypto = require('crypto')
+const fetch = require('node-fetch')
+const uuid = require('uuid/v4')
+function base64 (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+function hash (secret) { return crypto.createHash('sha256').update(secret).digest() }
+function hmac (secret, input) { return crypto.createHmac('sha256', secret).update(input).digest() }
+
+const plugin = new Plugin({
+  server: 'btp+ws://:@localhost:9000/'
+})
+
+let counter = 0
+
+function sendTransfer (obj) {
+  obj.id = uuid()
+  obj.from = plugin.getAccount()
+  // to
+  obj.ledger = plugin.getInfo().prefix
+  // amount
+  // executionCondition
+  obj.expiresAt = new Date(new Date().getTime() + 1000000).toISOString()
+  // console.log('calling sendTransfer!',  obj)
+  return plugin.sendTransfer(obj).then(function () {
+    return obj.id
+  })
+}
+
+plugin.connect().then(function () {
+  console.log('plugin connected')
+  return fetch('http://localhost:8000/')
+}).then(function (inRes) {
+  inRes.body.pipe(process.stdout)
+  const payHeaderParts = inRes.headers.get('Pay').split(' ')
+  console.log(payHeaderParts)
+  // e.g. Pay: 1 test.crypto.xrp.asdfaqefq3f.26wrgevaew SkTcFTZCBKgP6A6QOUVcwWCCgYIP4rJPHlIzreavHdU
+  setInterval(function () {
+    const ilpPacketContents = {
+      account: payHeaderParts[1] + '.' + (++counter),
+      amount: '1',
+      data: ''
+    }
+    const fulfillment = hmac(Buffer.from(payHeaderParts[2], 'base64'), ilpPacketContents.account)
+    const condition = hash(fulfillment)
+    sendTransfer({
+      to: payHeaderParts[1],
+      amount: '1',
+      executionCondition: base64(condition),
+      ilp: base64(IlpPacket.serializeIlpPayment(ilpPacketContents))
+    }).then(function () {
+      // console.log('transfer sent')
+    }, function (err) {
+      console.error(err.message)
+    })
+  }, 1)
+})
+```
+
+To run it, use:
+
+```sh
+node ./tool2.js
+```
+
+## What you learned
+
+We used the ILP packet for the first time, talked about connectors, payments as chains of transfers, and used
+the `Pay` header, so that the shop could request payment from the tool that connects to it.
+
+We added a BTP-enabled ledger to the shop, so that our content consumption tool can receive letters faster.
```

Comment:
I think this is too much to introduce in one tutorial. I think it would be better to separate trustlines/BTP from the `Pay` header